src/lib/paginator/paginator.ts

```diff
@@ -0,0 +1,166 @@
+
+  /** Displayed set of page length options. Will be sorted and include current page length. */
+  _displayedPageLengthOptions: number[];
+
+  constructor(public _intl: MdPaginatorIntl) { }
+
+  ngOnInit() {
+    this._initialized = true;
+
+    // If no page length or options have been provided, use the list length as the only option.
+    // Otherwise if there is no page length but there are options, then use the first option.
+    if (this.pageLength == undefined) {
+      this.pageLength = this.pageLengthOptions.length != 0 ?
+          this.pageLengthOptions[0] :
+          this.listLength;
+    }
+
+    this._updateDisplayedPageLengthOptions();
+  }
+
+  /** Increments the current page index to the next page index if a next page exists. */
+  navigateToNextPage() {
+    if (!this.canNavigateToNextPage()) { return; }
+    this.currentPageIndex++;
+    this._sendPageEvent();
+  }
+
+  /** Decrements the current page index to the previous page index if a next page exists. */
+  navigateToPreviousPage() {
+    if (!this.canNavigateToPreviousPage()) { return; }
+    this.currentPageIndex--;
+    this._sendPageEvent();
+  }
+
+  /** Returns true if the user can go to the next page. */
+  canNavigateToPreviousPage() {
+    return (this.currentPageIndex - 1) >= 0 && this.pageLength != 0;
+  }
+
+  /** Returns true if the user can go to the next page. */
+  canNavigateToNextPage() {
+    const numberOfPages = Math.ceil(this.listLength / this.pageLength) - 1;
+    return (this.currentPageIndex + 1) <= numberOfPages && this.pageLength != 0;
+  }
+
+  /**
+   * Changes the current page length so that the first item displayed on the page will still be
+   * displayed using the new page length.
+   *
+   * For example, if the page length is 10 and on the second page (items indexed 10-19) then
+   * switching so that the page length is 5 will set the third page as the current page so
+   * that the 10th item will still be displayed.
+   */
+  _changePageLength(pageLength: number) {
+    // Current page needs to be updated to reflect the new page length. Navigate to the page
+    // containing the previous page's first item.
+    const startIndex = this.currentPageIndex * this.pageLength;
+    this.currentPageIndex = Math.floor(startIndex / pageLength) || 0;
+
+    this.pageLength = pageLength;
+    this._sendPageEvent();
+  }
+
+  /**
+   * Updates the list of page length options to display to the user. Includes making sure that
+   * the current page length is an option and that the list is sorted.
+   */
+  private _updateDisplayedPageLengthOptions() {
+    if (!this._initialized) { return; }
+
+    this._displayedPageLengthOptions = this.pageLengthOptions.slice();
+    if (this._displayedPageLengthOptions.indexOf(this.pageLength) == -1) {
+      this._displayedPageLengthOptions.push(this.pageLength);
+    }
+
+    // Sort the numbers using a number-specific sort function.
+    this._displayedPageLengthOptions.sort((a, b) => (a - b));
```

Comment:
Unnecessary parentheses here: `this._displayedPageLengthOptions.sort((a, b) => a - b);`